### PR TITLE
consent/FHIR-34322: adding consent provision section and respective diagrams.

### PR DESCRIPTION
**Pull request by mojitoj**
## HL7 FHIR Pull Request

https://jira.hl7.org/browse/FHIR-34322

## Description

adding consent provision section and respective diagrams.

### CODE REVIEW

**Comment by mojitoj** [comment]
@DavidPyke I'm wondering why I'm getting a resource validation error here. My PR is purely documentation doesn't touch the examples or schema.

**Comment by mojitoj** [comment]
@lmckenzi We were wondering if we could get some help about the failing CI here. This PR only touches some documentation page but it seems to be failing on validating examples from another resource. I thought this might be an issue on `HEAD` but after leaving this here for almost two weeks I don't see those errors showing up other PRs.

**Comment by JohnMoehrke** [comment]
Builds local for me. 
does change a few files that are not part of this PR - ./source/patient/codesystem-pronouns.xml, ./source/patient/valueset-pronouns.xml, ./source/spelling/add.txt, ./tools/tx/snomed/snomed.xml

File: source/patient/valueset-pronouns.xml [hunk 1]
```diff
@@ -22,26 +27,26 @@
...
-	  </concept>
-	  <concept>
-		<code value="other"/>
-		<display value="Other"/>
-	  </concept>
-	  <concept>
-		<code value="unknown"/>
-		<display value="Unknown"/>
-	  </concept>
+      <concept>
```
Comment (mojitoj) [inline source/patient/valueset-pronouns.xml:30]: @JohnMoehrke are these added here intentionally? they don't seem to be in the scope of this ticket.

Comment (JohnMoehrke) [reply source/patient/valueset-pronouns.xml:30]: Some files get updated by the build process. When git notices these have changed, you are expected to include them. This is just part of the process of maintaining the fhir core.

Comment (JohnMoehrke) [reply source/patient/valueset-pronouns.xml:30]: yes, sometimes one finds that they change for reasons outside your scope. This is likely due to the previous person not submitting them properly... we all must work as a team.

**Comment by JohnMoehrke** [comment]
@grahamegrieve there seems to be an overly aggressive example validation complaining about DICOM and SNOMED codes.

**Comment by JohnMoehrke** [comment]
https://chat.fhir.org/#narrow/stream/179165-committers/topic/CI.20fails.20with.20errors.20unrelated.20to.20the.20PR

**Comment by Quarz0** [comment]
I wonder why this PR was merged when it adds explanation to the provision model of R4 that was already discussed as ambiguous and complex in https://jira.hl7.org/browse/FHIR-31412? Aren't alternative models being considered?

**Pull request merged**